File: backend/src/main/java/NetworkStandard/Messages/ReplayMessage.java
package NetworkStandard.Messages;


import NetworkStandard.Characters.CharacterInformation;
import NetworkStandard.DataTypes.MatchConfig.Matchconfig;
import NetworkStandard.DataTypes.Szenario.Scenario;
import NetworkStandard.MessageContainer.MessageContainer;
import NetworkStandard.MessageTypeEnum.MessageTypeEnum;

import java.util.Date;
import java.util.UUID;

/**
 * OPTIONAL
 * Message send by the server initialized by a client request.
 *
 * @author Standardisierungsdokument & Sedat Qaja
 */
public class ReplayMessage extends MessageContainer {
    // Metadaten
    UUID sessionId;
    Date gameStart;
    Date gameEnd;
    UUID playerOneId;
    UUID playerTwoId;
    String playerOneName;
    String playerTwoName;
    Integer rounds;
    // Konfigurationsdaten
    Scenario level;
    Matchconfig settings;
    CharacterInformation[] character_settings;
    // Alle Nachrichten
    MessageContainer[] messages;

    public ReplayMessage(UUID playerId, MessageTypeEnum type, Date creationDate, String debugMessage, UUID sessionId, Date gameStart, Date gameEnd, UUID playerOneId, UUID playerTwoId, String playerOneName, String playerTwoName, Integer rounds, Scenario level, Matchconfig settings, CharacterInformation[] character_settings, MessageContainer[] messages) {
        super(playerId, MessageTypeEnum.REPLAY, creationDate, debugMessage);
        this.sessionId = sessionId;
        this.gameStart = gameStart;
        this.gameEnd = gameEnd;
        this.playerOneId = playerOneId;
        this.playerTwoId = playerTwoId;
        this.playerOneName = playerOneName;
        this.playerTwoName = playerTwoName;
        this.rounds = rounds;
        this.level = level;
        this.settings = settings;
        this.character_settings = character_settings;
        this.messages = messages;
    }
}

File: dummy-ai-client/NetworkStandard/Messages/replay_message.py
from NetworkStandard.MessageContainer.message_container import MessageContainer
from NetworkStandard.MessageTypeEnum.messagetype_enum import MessageTypeEnum
from NetworkStandard.DataTypes.Matchconfig.matchconfig import Matchconfig
from NetworkStandard.DataTypes.Szenario.scenario import Scenario

import uuid


class ReplayMessage(MessageContainer):
    def __int__(self, player_Id, type_enum, creation_date, debug_message,
                session_Id,
                game_start, game_end, player_one_Id, player_two_Id, player_one_name,
                player_two_name, rounds, level, settings, character_settings,
                messages):
        super().__init__(player_Id, type_enum, creation_date, debug_message)
        self.session_Id = session_Id
        self.game_start = game_start
        self.game_end = game_end
        self.player_one_Id = player_one_Id
        self.player_two_Id = player_two_Id
        self.player_one_name = player_one_name
        self.player_two_name = player_two_name
        self.rounds = rounds
        self.level = level
        self.settings = settings
        self.character_settings = character_settings
        self.messages = messages
